### ISSUE TITLE
Regression in 3.6.6 in cancellation route

### ISSUE THREAD

**Issue by anuchandy** [reactor/reactor-core#3839]
In 3.4.x- the upstream cancellation hooks used to get invoked if the downstream happens to cancel the chain. For example, the below code, terminates an infinite Flux using take operator and the cancel signal used to arrive in the hook resulting the output _Cancelled_.  But in 3.6.6, this signal no longer arrives.

```java
    @Test
    public void testCancellation() {
        final Flux<Integer> flux = Flux.never();
        flux.doFinally(s -> {
            if (s == SignalType.CANCEL) {
                System.out.println("Cancelled");
            }
        }).take(0).blockLast();
    }
```

**Comment by chemicL** [comment]
Hey @anuchandy ! What you observe is a result of #2699 and was addressing #2698. It was originally released in 3.3.17.RELEASE, which is out of support. In the 3.4.x line, it was introduced in the [3.4.6](https://github.com/reactor/reactor-core/releases/tag/v3.4.6) version.

Given that:
* [3.4.x is going out of OSS support in August](https://projectreactor.io/docs/core/release/reference/#_support_timeline),
* At the time it was introduced, there were no considerations that this is a regression, but the change was introduced as an expected improvement.

I am sorry to say that I don't believe we will address your concern. The meaning of `take(0)` doesn't imply cancellation. I don't foresee a requirement like that to be fulfilled as it is counter-productive. As the javadoc explains, take(N) is equivalent to `take(N, limitRequest=true)`. A replacement by `take(0, false)` will yield the expected result on the other hand. Please consider using that variant in your case.

**Comment by anuchandy** [comment]
Hello @chemicL, thanks for checking and confirming that this is expected in 3.6.x. We will update our tests to use the alternate api.
